templates/commands/goldentest/verify.go

```diff
@@ -124,40 +148,50 @@ func verify(ctx context.Context, templateLocation string, testNames []string, st
 		green = fmt.Sprint
 	}
 
-	var merr error
-	var resultReport strings.Builder
+	for _, tc := range testCases {
+		tc := tc
 
-	resultReport.WriteString("\nTest Report:\n")
+		workerFunc := func() (_ string, rErr error) {
+			tempTracker := tempdir.NewDirTracker(fs, false)
+			defer tempTracker.DeferMaybeRemoveAll(ctx, &rErr)
 
-	for _, tc := range testCases {
-		p := &runTestCaseParams{
-			templateLocation: templateLocation,
-			tempBase:         tempDir,
-			useColor:         useColor,
-			redSprintf:       red,
-			greenSprintf:     green,
-		}
-		if err := runTestCase(ctx, p, tc); err != nil {
-			result := red(fmt.Sprintf("[x] template location [%s] golden test [%s] fails", templateLocation, tc.TestName))
-			tcErr := fmt.Errorf("%s:\n %w", result, err)
-			merr = errors.Join(merr, tcErr)
-			resultReport.WriteString(result)
-		} else {
-			resultReport.WriteString(green(fmt.Sprintf("[✓] template location [%s] golden test [%s] succeeds", templateLocation, tc.TestName)))
-		}
+			tempDir, err := tempTracker.MkdirTempTracked("", "verify-testcase-"+tc.TestName+"-")
```

Comment:
just curious why we need a `-` at the end?

Reply:
It will be appended with a random cookie, like `verify-testcase-foo-123123`. It's nicer to have the dash to keep the parts separate.

https://pkg.go.dev/os#MkdirTemp

Reply:
Thank you! Learned another new thing from you :)